### PR TITLE
expect fx codegen

### PR DESCRIPTION
**Pull request by folkertdev**
- boilerplate
- more boilerplate
- thread fx expects to the runner
- expect-fx runner implementation
- get expect-fx to actually run

### CODE REVIEW

**Comment by folkertdev** [comment]
CI fails in a weird way. I'll re-run and see if it's a flake or real

```
---- cli_run::helloWorld stdout ----
thread 'cli_run::helloWorld' panicked at '`valgrind` exited with no exit code. valgrind stdout was: "Hello, World!
"

valgrind stderr was: ""', crates/cli/tests/cli_run.rs:206:29
stack backtrace:
   0: rust_begin_unwind
             at /rustc/fe5b13d681f25ee6474be29d748c65adcd91f69e/library/std/src/panicking.rs:584:5
   1: core::panicking::panic_fmt
             at /rustc/fe5b13d681f25ee6474be29d748c65adcd91f69e/library/core/src/panicking.rs:143:14
   2: cli_run::cli_run::check_output_with_stdin
   3: core::ops::function::FnOnce::call_once
   4: core::ops::function::FnOnce::call_once
             at /rustc/fe5b13d681f25ee6474be29d748c65adcd91f69e/library/core/src/ops/function.rs:227:5
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.

---- cli_run::platformSwitching stdout ----
thread 'cli_run::platformSwitching' panicked at '`valgrind` exited with no exit code. valgrind stdout was: "Which platform am I running on now?
"

valgrind stderr was: ""', crates/cli/tests/cli_run.rs:206:29
stack backtrace:
   0: rust_begin_unwind
             at /rustc/fe5b13d681f25ee6474be29d748c65adcd91f69e/library/std/src/panicking.rs:584:5
   1: core::panicking::panic_fmt
             at /rustc/fe5b13d681f25ee6474be29d748c65adcd91f69e/library/core/src/panicking.rs:143:14
   2: cli_run::cli_run::check_output_with_stdin
   3: core::ops::function::FnOnce::call_once
   4: core::ops::function::FnOnce::call_once
             at /rustc/fe5b13d681f25ee6474be29d748c65adcd91f69e/library/core/src/ops/function.rs:227:5
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.

---- cli_run::ruby stdout ----
thread 'cli_run::ruby' panicked at '`valgrind` exited with no exit code. valgrind stdout was: "I need a string here!"

valgrind stderr was: ""', crates/cli/tests/cli_run.rs:206:29
stack backtrace:
   0: rust_begin_unwind
             at /rustc/fe5b13d681f25ee6474be29d748c65adcd91f69e/library/std/src/panicking.rs:584:5
   1: core::panicking::panic_fmt
             at /rustc/fe5b13d681f25ee6474be29d748c65adcd91f69e/library/core/src/panicking.rs:143:14
   2: cli_run::cli_run::check_output_with_stdin
   3: core::ops::function::FnOnce::call_once
   4: core::ops::function::FnOnce::call_once
             at /rustc/fe5b13d681f25ee6474be29d748c65adcd91f69e/library/core/src/ops/function.rs:227:5
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.


failures:
    cli_run::helloWorld
    cli_run::platformSwitching
    cli_run::ruby

test result: FAILED. 35 passed; 3 failed; 0 ignored; 0 measured; 0 filtered out; finished in 295.39s
```

**Comment by Anton-4** [comment]
I'm able to reproduce the helloWorld and platformSwitching test errors on NixOS but not the ruby one.
I'll continue investigating.

**Comment by Anton-4** [comment]
```
valgrind --leak-check=full --track-origins=yes ./helloWorld 
==336391== Memcheck, a memory error detector
==336391== Copyright (C) 2002-2022, and GNU GPL'd, by Julian Seward et al.
==336391== Using Valgrind-3.19.0 and LibVEX; rerun with -h for copyright info
==336391== Command: ./helloWorld
==336391== 
Hello, World!
==336391== 
==336391== Process terminating with default action of signal 11 (SIGSEGV): dumping core
==336391==  Bad permissions for mapped region at address 0x120074
==336391==    at 0x120074: ??? (in /home/anton/gitrepos/roc3/roc/examples/hello-world/helloWorld)
==336391==    by 0x4981204: __run_exit_handlers (in /nix/store/scd5n7xsn0hh0lvhhnycr9gx0h8xfzsl-glibc-2.34-210/lib/libc.so.6)
==336391==    by 0x4981389: exit (in /nix/store/scd5n7xsn0hh0lvhhnycr9gx0h8xfzsl-glibc-2.34-210/lib/libc.so.6)
==336391==    by 0x496B23D: (below main) (in /nix/store/scd5n7xsn0hh0lvhhnycr9gx0h8xfzsl-glibc-2.34-210/lib/libc.so.6)
==336391== 
==336391== HEAP SUMMARY:
==336391==     in use at exit: 0 bytes in 0 blocks
==336391==   total heap usage: 0 allocs, 0 frees, 0 bytes allocated
==336391== 
==336391== All heap blocks were freed -- no leaks are possible
==336391== 
==336391== For lists of detected and suppressed errors, rerun with: -s
==336391== ERROR SUMMARY: 0 errors from 0 contexts (suppressed: 0 from 0)
Segmentation fault (core dumped)
```
This possibly has the same cause as #3540, namely different versions of glibc being used together.

File: crates/repl_expect/Cargo.toml [hunk 1]
```diff
@@ -26,10 +26,11 @@ roc_build = { path = "../compiler/build" }
 
 libloading = "0.7.1"
 inkwell = { path = "../vendor/inkwell" }
+signal-hook = "0.3.14"
+libc = "0.2.106"
 
 [dev-dependencies]
 test_gen = { path = "../compiler/test_gen" }
-libc = "0.2.106"
```
Comment (Anton-4) [inline crates/repl_expect/Cargo.toml:32]: This change may be the cause.

**Comment by Anton-4** [comment]
Changes made because of the zig-overlay have probably fixed our earlier issue (for now).

**Comment by rtfeldman** [comment]
yesssss, so excited to see this green! 😻 😻 😻

**Review by rtfeldman — APPROVED** [review]

**Pull request merged**